metacat-connector-druid/src/main/java/com/netflix/metacat/connector/druid/client/DruidHttpClientImpl.java

```diff
@@ -0,0 +1,105 @@
+/*
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.connector.druid.client;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.netflix.metacat.common.exception.MetacatNotFoundException;
+import com.netflix.metacat.common.json.MetacatJsonLocator;
+import com.netflix.metacat.common.server.connectors.ConnectorContext;
+import com.netflix.metacat.connector.druid.DruidConfigConstants;
+import com.netflix.metacat.connector.druid.MetacatDruidClient;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * DruidHttpClientImpl.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Slf4j
+public class DruidHttpClientImpl implements MetacatDruidClient {
+    private String druidURI;
+    private final RestTemplate restTemplate;
+    private final MetacatJsonLocator jsonLocator = new MetacatJsonLocator();
+
+    /**
+     * Constructor.
+     *
+     * @param connectorContext connector context
+     * @param restTemplate     rest template
+     */
+    public DruidHttpClientImpl(final ConnectorContext connectorContext,
+                               final RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+        final Map<String, String> config = connectorContext.getConfiguration();
+        final String host = config.get(DruidConfigConstants.DRUID_SERVER);
+        final String port = String.valueOf(config.get(DruidConfigConstants.DRUID_PORT));
+        final String coordinatorUri = config.computeIfAbsent(DruidConfigConstants.DRUID_COORDINATOR_URI,
+            p -> DruidConfigConstants.DRUID_COORDINATOR_ENDPOINT);
+
+        if (host != null && port != null) {
```

Comment:
I would flip this check. Make an `if` with or conditions for speed and throw the exception if they match. Then you can just assign the `druidURI` normally after. Also doesn't seem like a `NotFoundException` seems more like a bad configuration type exception to me since you can't even start it up. I would throw a not found exception if you can't actually connect to the server which is maybe something else you want to check in here during bootstrapping?